### ISSUE TITLE
Memphis - Usage not showing in SABilling

### ISSUE THREAD

**Issue by kdjstudios** [kdjstudios/SABillingGitlab#1746]
In GitLab by @kdjstudios on Oct 16, 2019, 10:21

**Submitted by:** "Laura Duckworth" <laura.duckworth@answernet.com>

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2019-10-16-63162

**Helpdesk:** http://www.servicedesk.answernet.com/profiles/ticket/2019-10-16-49134

**Server:** Internal

**Client/Site:** Memphis

**Account:** Multiple

**Issue:** 

Memphis, TN

Belz Investco, LTD

044-4858 

SABilling is not logging Usage History for this account.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2019, 10:21

I wrote:

> We just checked the account's HR report and it is showing their is usage on the account. May you please explain in further detail what is missing?

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2019, 11:00

changed title from **Memphis - {-Belz Investco, LTD -}Usage not showing in SABilling** to **Memphis - Usage not showing in SABilling**

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2019, 11:00

changed the description

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2019, 11:01

Laura wrote:

> Memphis, TN 
> 
> Otolaryngology Consultants
> 
> 044-5250 
> 
> ENT of Memphis / Otolaryngology Consultants
> 
>  
> 
> Usage history not showing in SABilling for this account

I wrote:

> The Usage history report only reports for invoiced items. So if it was not invoiced it will not show. Also I do not believe SA Billing will invoice for zero based charges. So if it was not charged it would not show in the Usage history report. 
> 
> NOTE: The HR report displays all usage metrics uploaded. So that may be the report you may want to review if you looking for zero based charges. 

Laura wrote:

> The Usage History area should show a running tab month over month of how many calls/ mins/whatever item the account has for an allowance regardless if it's over the allowance or not... this account has none at all .... I attached an example 
![Healthcare+Trust](/uploads/d1ba71a1e1466143fcc79b8f6872fa7b/Healthcare+Trust.PNG)

I wrote:

> To my understanding the usage history will show if those charges are active and being invoiced, thus why on this account they are showing. From a quick look at the account Belz Investco last few invoices, we do not see any usage based charges on their invoices, thus why they is no usage history. 
> 
> We will have the support team confirm this and follow up with their response.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Oct 16, 2019, 11:02

@amishra @smasih @jredfield 

So it looks like the issue is that the accounts are showing usage via the HR report, but are not showing any usage on the usage history report. Also looks like there has been no usage charges on the account. Which I presume is then why it does not display on the usage history. Please look into what is causing usage to not be charged, which in turn causes it to not be on the usage history report.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Nov 28, 2019, 04:07

@kdjstudios @jredfield 

Thank you for the update.

* We dug into the code and our finding describes below.

* This account has turn off date `02/01/2018` and Turn on date `06/01/2018`
* The turn on date should be less than the turn off date. As you can see the turn on date is greater than the turn of date.that's why you are not able to see the usage history.

* These scenarios same as ticket #1064 and to solve this issue we are tracking the account activation and reactivation as suggested by Tim's [comment](http://gitlab.aavaz.biz/AnswerNet/SABilling/issues/1064#note_44647).

* We had added an entry in the new account termination model if `trun_of_date` is present on the account. after that user reactive the account we had not created an entry in the new account termination model.

* As per the current functionality if the User reactivates the account then the `turn_of_date` set as `blank`.

* **In short**, to resolve the usage history report,  we need to run the script and that script will add one new entry into the account termination model("Reactivated") and set `turn_of_date` as a blank. this script will helpful for the on-off report as well.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Nov 29, 2019, 11:38

@amishra 

Please post an estimate to run that script to fix this. also should it be 'turn_off_date' looks like you misspelled it. 

@jredfield 

Should we be filtering this report by turn on and off dates even? I feel like we should be showing all data, regardless of the dates. That way ops will see it and can also determine if they missed billing usage, because they termed or started an account at the wrong date.

**Comment by kdjstudios** [comment]
In GitLab by @ranand on Dec 2, 2019, 03:53

@kdjstudios @jredfield 

## Developer Estimate
|Size|Lower Time Bound|Variability|Upper Time Bound|Testing|Time for Testing| Time for Dev & Testing|
|-|-|-|-|-|-|-|
|R-Small|2|Concepts and Implementation Well Understood|3|Low Impact|1|4|

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Dec 2, 2019, 06:10

@ranand Kyle wrote

> Should we be filtering this report by turn on and off dates even? I feel like we should be showing all data, regardless of the dates. That way ops will see it and can also determine if they missed billing usage, because they termed or started an account at the wrong date.

I am ready to approve, but see that this question not really answered. @kdjstudios this would be our decision, right? Once this is decided, I will approve.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 2, 2019, 14:56

@jredfield 

That question is for you to answer. You may want to reference with Gary's input if you do not know.

**Comment by kdjstudios** [comment]
In GitLab by @jredfield on Dec 4, 2019, 05:41

@kdjstudios wrote:

> Should we be filtering this report by turn on and off dates even? I feel like we should be showing all data, regardless of the dates.

Yes please show all data regardless of dates.  Approved.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 4, 2019, 10:07

@jredfield 

The estimate was to fix the existing accounts who have 'bad' turn off dates. This is because due to a bug in the past when they were reactivated it did not clear out the turn off date. This still needs to be fix, so I am presuming that is what you are approving.  

In regards to updating the usage report to show all dates, lets create a new ticket and get Gary's input. #1647, I have a feeling once we have the above fixed, we may not need to change this report.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 5, 2019, 02:41

assigned to @amishra

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 5, 2019, 04:07

@kdjstudios @jredfield 

We have run the script on the dev server and usage history report is showing and also create one entry into account termination tracker with status "Reactivated" and date with account turn on date. we can check from the account ledger page.

Please look into this and confirm if we can run same on the both of the production servers.

![image](/uploads/286487512cca911af9f134e0021a4cec/image.png)

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 5, 2019, 11:17

@amishra 

Looks good to me. If we are able to run this during this weekend release would be wonderful.

**Comment by kdjstudios** [comment]
In GitLab by @amishra on Dec 9, 2019, 24:39

@kdjstudios @jredfield 


Thank you for the update.

We have successfully released on both of the production servers. Please have a look.

Thanks

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 9, 2019, 09:00

Thanks. marking for close.

**Comment by kdjstudios** [comment]
In GitLab by @kdjstudios on Dec 9, 2019, 09:00

closed
